src/cpp/configuration/DDSRouterConfiguration.cpp

```diff
@@ -102,6 +102,7 @@ std::list<std::shared_ptr<FilterTopic>> DDSRouterConfiguration::generic_get_topi
             {
                 std::string new_topic_name;
                 std::string new_topic_type;
+                bool new_topic_with_key = false;
```

Comment:
NIT: this is a dark practice, assuming the default value of a parameter is dangerous.
However, the way to do it is a ugly and long, so I would not worry about it more than adding a comment.

Reply:
:+1: